Documentation updates!

ryansurf opened the pull request:
@K-dash 

This PR makes changes to the documentation.

The following changes have been made:
- The addition of the `cheet_sheet.md` file. This list typical commands that I run on the daily, and may be helpful for someone who is learning Git/the codebase
- The addition of a Discord widget and a widget of the repo in `index.md`
- Updated project structure in `structure,md`
- Test doc update, including how to run `make test` and a screenshot of the coverage report
- Added the new `cheet_sheet.md` to the `mkdocs.yml` file

I sent you an invite to be a collaborator, so I think you should be able to merge this PR. Let me know if you have any issues!

K-dash commented on docs/structure.md:
```diff
@@ -36,28 +39,38 @@ More in-depth structure:
+│   ├── settings.py
 │   ├── static
 │   └── templates
 ├── start_venv.sh
```
With the introduction of Poetry, I think this file is no longer being used, so it might be okay to delete it!

K-dash replied:
While it's not necessary now, if the number of project files increases in the future, modifying this file could become a hassle. So, ex.g it might be a good idea to add a simple script to `pre-commit` that automatically retrieves and reflects the directory structure :)

ryansurf replied:
> With the introduction of Poetry, I think `start_venv.sh` is no longer being used, so it might be okay to delete it!

Ah, forgot about that. Good call

ryansurf replied:
> While it's not necessary now, if the number of project files increases in the future, modifying this file could become a hassle. So, ex.g it might be a good idea to add a simple script to `pre-commit` that automatically retrieves and reflects the directory structure :)
> 
> 

How did I not think of this...good idea. I'll work on getting that going

K-dash commented on docs/cheat_sheet.md:
```diff
@@ -0,0 +1,52 @@
+# Cheat Sheet!
```
Nice!

K-dash reviewed (commented):
@ryansurf 
I have reviewed the changes! Please take a look :)

ryansurf commented:
> @ryansurf 
> 
> I have reviewed the changes! Please take a look :)

Thank you! I'll implement the changes you recommended and submit another PR

K-dash commented:
> Thank you! I'll implement the changes you recommended and submit another PR

You're welcome!

(Please ignore this if you already know 😃)
In the usual review process, you would create a fix commit in the same PR and push it to show that you have addressed the feedback. However, if the fix is expected to take time or can be handled later, the workflow is to create a separate PR.
In this case, I judged that the latter seems applicable, so for now, I will go ahead and merge this PR :)

ryansurf commented:
> > Thank you! I'll implement the changes you recommended and submit another PR
> 
> You're welcome!
> 
> (Please ignore this if you already know 😃) In the usual review process, you would create a fix commit in the same PR and push it to show that you have addressed the feedback. However, if the fix is expected to take time or can be handled later, the workflow is to create a separate PR. In this case, I judged that the latter seems applicable, so for now, I will go ahead and merge this PR :)

I see, thank you for the merge 😄
